sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -620,7 +621,7 @@ abstract class UnixTime
   override def inputTypes: Seq[AbstractDataType] =
     Seq(TypeCollection(StringType, DateType, TimestampType), StringType)
 
-  override def dataType: DataType = LongType
+  override def dataType: DataType = DoubleType
```

Comment:
BTW, I think we can't just change this datatype directly. This could break backward compatibility.